### PR TITLE
Raydium CLMM

### PR DESCRIPTION
**Pull request by brittcyr**
Also search through raydium clmm in addition to orca whirlpools

### CODE REVIEW

**Comment by brittcyr** [comment]
Example:
curl 'http://localhost:5000/swap?wallet=Bz9thGbRRfwq3EFtFtSKZYnnXio5LXDaRgJDh3NrMAGT&inputMint=EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v&outputMint=4k3Dyjzvzp8eMZWUXbBCjEvwSkkk59S5iCNLY3QrkX6R&mode=ExactIn&amount=100000000&slippage=0.001' | jq

[
  {
    "amount": "100000000",
    "otherAmountThreshold": "0",
    "mode": "ExactIn",
    "slippage": 0.001,
    "inAmount": "100000000",
    "outAmount": "541041203",
    "priceImpact": 0.001709826033485129,
    "marketInfos": [
      {
        "label": "Whirlpool",
        "fee": {
          "amount": "20000",
          "mint": "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v",
          "rate": 0.00019999999999999998
        }
      },
      {
        "label": "Raydium",
        "fee": {
          "amount": "11372536",
          "mint": "So11111111111111111111111111111111111111112",
          "rate": 0
        }
      }
    ],
    "mints": [
      "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v",
      "So11111111111111111111111111111111111111112",
      "4k3Dyjzvzp8eMZWUXbBCjEvwSkkk59S5iCNLY3QrkX6R"
    ],
...

File: src/router.ts [hunk 1]
```diff
@@ -199,19 +203,173 @@ class WhirlpoolEdge implements Edge {
...
+        ok: ok,
+        instructions,
+        label: this.poolPk.toString(),
+        marketInfos: [
+          {
+            label: "Raydium",
+            fee: {
+              amount: fee,
+              mint: this.inputMint,
+              rate: 0, // TODO: Fix this
```
Comment (mschneider) [inline src/router.ts]: the rate is displayed in the UI, would be good if this can be displayed

File: package.json [hunk 2]
```diff
@@ -18,12 +18,15 @@
...
     "@project-serum/anchor": "^0.26.0",
+    "@raydium-io/raydium-sdk": "^1.3.1-beta.8",
     "@solana/web3.js": "^1.70.1",
     "@types/bn.js": "^5.1.1",
     "@types/node": "^18.11.18",
+    "bs58": "^5.0.0",
     "cors": "^2.8.5",
     "express": "^4.18.2",
     "express-prom-bundle": "^6.6.0",
+    "js-sha256": "^0.9.0",
```
Comment (mschneider) [inline package.json]: as this is running on node, probably better to use a polyfill that resolves to node-crypto like "sha256", or node crypto directly

Comment (mschneider) [reply package.json]: anchor uses crypto-hash internally

Comment (brittcyr) [reply package.json]: looks like anchor uses @noble/hashes/sha256

**Review by mschneider — APPROVED** [review]

**Pull request merged**